sstables/storage.cc

```diff
@@ -111,16 +111,15 @@ future<file> filesystem_storage::open_component(const sstable& sst, component_ty
     auto name = sst.filename(tgt_dir, type);
 
     auto f = open_sstable_component_file_non_checked(name, flags, options, check_integrity);
+    if (tgt_dir == dir) {
```

Comment:
This looks like slower check for me -- it's effectively the strcmp. While it was a bool-flag check. The `temp_dir` can be unset only in some rare corner cases (if at all), so I'd keep the existing check even if it loses some cycles in false-positive case.

Reply:
okay. i don't want to insist. but the `strcmp` versus bool-flag check is not the key here. the point is that `tgt_dir == dir` is more correct.